### PR TITLE
Complex Builds MVP (Crust): Get main port # from docker container.

### PR DESCRIPTION
**Pull request by OAGr**
This does the following:
A) fix the 'launch environment' button
B) Get the Port for an environment
C) Add the 'Delete' button for everything but building builds (and refactor the other buttons a bit)
D) small fix for X-Fleet (Global-case)

### CODE REVIEW

File: lib/app/views/_build_info.erb [hunk 1]
```diff
@@ -24,37 +24,39 @@
...
+        <input name="template" type="hidden" value="server"/>
+        <button class="btn btn-default" title='Launch Server' type='submit'>
+          Launch Server
+        </button>
+      </form>
+
       <% if build.completed? or build.service_template != 'spec' %>
         <form action="<%= build_path(build) %>" method="post" class="submit">
-          <button class="btn btn-default" title='Rebuild this commit' type='submit'>
+          <button class="btn btn-default" title='Launch Specs' type='submit'>
```
Comment (rahmal) [inline lib/app/views/_build_info.erb]: Still think *Launch* is ambiguous. Can we make it *Build* or *Run*? As in ***Build Specs/Build Server*** or ***Run Specs/Run Server***. IMHO *Run* is the better choice of the three.

File: lib/integrity/build.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ class Build
     property :fleetctl_status,       String
     property :machine_ip,            String
     property :service_template,      String, default: 'spec'
+    property :public_port,           String
```
Comment (rahmal) [inline lib/integrity/build.rb]: This is only the public port for jarvis correct? Are you parsing the ports for the other apps? DJ has none, but what about MailCatcher, and Frick Later? It's not required right now, but soon we will want the ports for all apps, so keep it in mind. We may need to make this a json field.

File: lib/integrity/build.rb [hunk 3]
```diff
@@ -149,6 +163,10 @@ def notify
       project.enabled_notifiers.each { |n| n.notify(self) }
     end
 
+    def reload
+      Build.get(id)
+    end
```
Comment (rahmal) [inline lib/integrity/build.rb]: should be ```self.class.get(id)```. This seems harmless, but when you are calling a class method on the instances own class you should get in the habit of using self.class. That way in cases where the class might be subclassed or the method overridden, or something similar, it's not hardcoded always use the build class. 

This was once the cause of an infamous bug in Rails 2. I'd hate to be the guy who wrote that. ;)

File: lib/integrity/builder.rb [hunk 4]
```diff
@@ -28,8 +28,9 @@ def self.queue_next_build
     def self.launch_build(build)
       log "Launching Build: #{build.service_vars.inspect}"
       build.cleanup!
-      # build.rebuild ? build.cleanup! : build.kill_previous_builds!
       Crust.start(*(build.service_vars + [build.service_template]))
+      Build.update_fleetctl_status
+      build = build.reload
```
Comment (rahmal) [inline lib/integrity/builder.rb]: Lines can be collapsed into ```build.reload.started!```. Less code and still very readable.

File: services/logspout.service [hunk 5]
```diff
@@ -12,5 +12,5 @@ ExecStart=/usr/bin/docker run -v=/var/run/docker.sock:/tmp/docker.sock grnds/log
 ExecStop=/usr/bin/docker kill logspout
 ExecStopPost=/usr/bin/docker ps -a -q | xargs docker rm
 
-[X-fleet]
+[X-Fleet]
```
Comment (rahmal) [inline services/logspout.service]: Wow. Really? Fleet sucks for that.

**Comment by rahmal** [comment]
CR'd. Looks great so far! Please fix minor comments and push it up.

File: lib/integrity/builder.rb (hunk 4 above)
Comment (OAGr) [inline lib/integrity/builder.rb]: Good point.

File: lib/integrity/build.rb (hunk 2 above)
Comment (OAGr) [inline lib/integrity/build.rb]: We'll add the mailcatcher and others later.  Wanted to start with one. 

That's a good point about the json field though, I was worried about the number of them.

File: lib/app/views/_build_info.erb (hunk 1 above)
Comment (OAGr) [inline lib/app/views/_build_info.erb]: I'll change it to Run.

File: services/logspout.service (hunk 5 above)
Comment (OAGr) [inline services/logspout.service]: the source of many bugs for us so far

**Pull request merged**